packages/dom/src/dom.js

```diff
@@ -130,14 +130,21 @@ export function isHorizontalEdge( container, isReverse ) {
 	// If confirmed to be at extent, traverse up through DOM, verifying that
 	// the node is at first or last child for reverse or forward respectively.
 	// Continue until container is reached.
-	const order = isReverse ? 'first' : 'last';
+	const order = isReverse ? 'previous' : 'next';
```

Comment:
I'll have a look into them. Selection set by rich text is purely done with text nodes, so sometimes we need these trailing text nodes. I can see if we con do it without them.

Reply:
Just noting, this is a workaround for a browser bug. Selection can never be set to the empty text node by arrow nav, so you can never reach the end.

Reply:
As this is technically a bug in master as well (the empty text nodes have been there for a while) and it would require changes to selection setting, I think it would be better to look into separately. I don't think adjusting selection setting will be trivial.

In any case, this seems like good behaviour to add to me. Visually the caret is at the end, so the it's at the horizontal edge.

Reply:
We should still update the comment then, to explain the exceptions we're considering. Right now, the comment doesn't reflect reality.

>verifying that the node is at first or last child for reverse or forward respectively.